src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventPipe.cs

```diff
@@ -177,7 +178,7 @@ internal void SetProfilerSamplingRate(TimeSpan minTimeBetweenSamples)
 
     internal static class EventPipe
     {
-        private static ulong s_sessionID = 0;
+        internal static ConcurrentQueue<ulong> s_sessionIDs = new ConcurrentQueue<ulong>();
```

Comment:
Could we use a `HashSet` or just a `List` and expose the ability to disable a specific session or at least the number of active sessions?  This change currently expects users of this API to keep track of how many sessions they open, and more importantly, expects there to only be _one_ user.  If there are multiple callers to this API and they interleave calls to `Enable` and `Disable`, they could end up disabling each other's sessions.

e.g.,
1. Caller1 calls `Enable`
2. Caller2 calls `Enable`
3. Caller2 calls `Disable` and they just disabled Caller1's session

Reply:
Yup. I thought about this too, and discussed options available with @xiaomi7732 from AppInsights Service Profiler team. The problem with doing that is that it changes the method signature, which breaks back-compat with 2.2 which is still supported by AppInsights ServiceProfiler (which, afaik, is the only team that enables EventPipe using private reflection). 

It's not ideal and there is the case you mentioned which would ideally break. But customers shouldn't be (and were never encouraged by us via any blog articles, docs, etc.) to enable/disable EventPipe this way, so I'm hoping that people won't run into such issues, and when they do, changing their code to not use reflection will be actually easier than whatever they were doing before :)